src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const obj: GroupsMap<T> = {};
+
+  for (let i = 0; i < items.length; i++) {
+    const value = items[i][key] as string;
+
+    if (!obj.hasOwnProperty(value)) {
+      obj[value] = [];
+      obj[value].push(items[i]);
```

Comment:
This line is redundant because the item is pushed to the array in both the `if` and `else` blocks. You can simplify the code by removing this line and moving the `push` operation outside of the `if-else` block.